DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Administrator/AdministratorController.cs

```diff
@@ -131,6 +132,54 @@ public IActionResult UnlockAccount(int adminId)
             return RedirectToAction("Index");
         }
 
+        [Route("{adminId:int}/DeactivateAdmin")]
+        [HttpGet]
+        public IActionResult DeactivateAdmin(int adminId)
+        {
+            var centreId = User.GetCentreId();
+            var adminUser = userDataService.GetAdminUserById(adminId);
+            if (adminUser == null || adminUser.CentreId != centreId)
+            {
+                return NotFound();
+            }
+
+            var model = new DeactivateAdminViewModel(adminUser);
+            return View(model);
+        }
+
+        [Route("{adminId:int}/DeactivateAdmin")]
+        [HttpPost]
+        public IActionResult DeactivateAdmin(int adminId, DeactivateAdminViewModel model)
+        {
+            if (!model.Confirm)
+            {
+                ModelState.AddModelError(
+                    nameof(DeactivateAdminViewModel.Confirm),
+                    "You must confirm before deactivating this account."
+                );
+                return View(model);
+            }
+
+            var centreId = User.GetCentreId();
+            userDataService.DeactivateAdmin(adminId, centreId);
+
+            return RedirectToAction(nameof(DeactivateAdminConfirmation), new { adminId });
+        }
+
+        [Route("{adminId:int}/DeactivateAdminConfirmation")]
+        [HttpGet]
+        public IActionResult DeactivateAdminConfirmation(int adminId)

```

Comment:
I think this can be refactored to be simpler. Seeing as how nothing account specific is displayed on the view, I don't think we need to take in an adminId here, we could just show the page as it is. In fact, we could probably get rid of this entire method, and just have the DeactivateAdmin Post method return the new view when it successfully deactivates the admin.

Reply:
This viewmodel class removed so action method also removed